bal_tools/subgraph.py

```diff
@@ -93,7 +99,10 @@ def get_subgraph_url_frontendv2(self, subgraph):
                         if urlparse(url).scheme in ["http", "https"]:
                             graph_api_key = os.getenv("GRAPH_API_KEY")
                             if "${keys.graph}" in url and not graph_api_key:
-                                warnings.warn(f"`GRAPH_API_KEY` not set. may be rate limited or have stale data for subgraph:{subgraph} url:{url}", UserWarning)
+                                warnings.warn(
+                                    f"`GRAPH_API_KEY` not set. may be rate limited or have stale data for subgraph:{subgraph} url:{url}",
+                                    UserWarning,
+                                )
                                 break
```

Comment:
Doesn't this break cause premature exit, discarding the URL even when a graph API key exists?

Reply:
the frontend only has prod urls. so if we dont have a key, it should fail (and move on the the next method of retrieving a url, in this case `get_subgraph_url_legacy`)

ref: eg https://raw.githubusercontent.com/balancer/frontend-v2/master/src/lib/config/arbitrum/index.ts